### DISCUSSION TITLE
Minimum requirements for a client

### DISCUSSION THREAD

**Discussion by upintheairsheep** [FreeRDP/FreeRDP#10600] [Q&A]
Hello, I want to try out seeing if FreeRDP can revive some ancient computers I have access to. What are the lowest requirements, especially the RAM needed to run freeRDP combined with the required RAM to run bare Linux with all the FreeRDP dependencies? Additionally, can this compile to run on any architecture such as x86 (32-bit), ARM32, PPC, PPC64, MIPS, SuperH, and m68k (all provided that I have enough RAM to run everything).

**Comment by hardening** [comment]
It kinda depends on the target resolution and the negotiated codecs, but if you run a "normal" connection (so with all options by default) you will quickly go to at least 100MB of caches in various places (with sizes set in the protocol, so there's not much we can do to limit their size). Add to that the size of 1 or 2 screens in 32 bits, and you may have a raw estimation.

**Comment by akallabeth** [comment]
Another datapoint:
I´ve just ran a connection to a windows 10 with `AVC444` mode, multimonitor, 2 monitors 3840x2160 resolution and the memory used was ~290MB

of more importance is most likely the CPU and memory bandwidth to your graphics card:
The whole RDP data must be decoded:
1. different video codecs with post processing (AVC444 for example is 2x H264 + a expensive YUV -> RGB recombination)
2. audio codecs (sound and microphone, PCM or AAC)
3. other protocol overhead

so your machine must be fast enough to:
1. decode/encode the whole stuff
2.  copy and display bitmaps of whatever resolution you are running in the framerate that they appear (I think it is limited to 33 FPS, not sure though)

as for architectures: https://github.com/FreeRDP/FreeRDP/actions/workflows/alt-architectures.yml
it should run on any architecture, even big endian machines (although your mileage there may vary, really not many testers)

**Comment by hardening** [comment]
Remote frame limitation: https://learn.microsoft.com/en-us/troubleshoot/windows-server/remote/frame-rate-limited-to-30-fps

**Reply by upintheairsheep** [reply]
What is the limit for 30 FPS, one monitor, HD resolution?

**Reply by akallabeth** [reply]
@upintheairsheep 
1. you mean `HD` as in `1280x720` or `Full HD` as in `1920x1080`?
2. for `30 FPS` your hardware needs to be (at least):
   * capable of decoding 2x H264 streams in your resolution (maybe a little less if you deactivate `AVC444`)
   * pushing 30 bitmaps from your system memory to your screen per second (again in your resolution)
   * some remaining margin for SSL encryption/decryption and network overhead

that is a rough estimation of the minimum required.

**Reply by upintheairsheep** [reply]
Just want a general minimum requirement for RAM. I plan to see if ancient computers, old phones, and old game consoles, can FreeRDP and a lightweight Linux distro be a client for a host device with 256 MB RAM? 128 MB? 512 MB?

**Reply by akallabeth** [reply]
@upintheairsheep then just start a session with your parameters and monitor the ram usage.
caches et al are allocated at session startup so there is not much variance during runtime (you still need a few `MB` spare for dynamic allocations)

the requirements do vary quite a bit depending on build parameters, but I´d say you need at least 200MB (rough estimation, better measure that yourself as we do not consider this something worthy of checking - as long as the overall usage is in the right magnitude)
